### PR TITLE
Fix failure of qvm_stabilizer_example.ipynb with numpy-2.4.0

### PR DESCRIPTION
**Pull request by pavoljuhas**
- Avoid conversion of a one-element numpy array to float which
  throws TypeError in numpy-2.4.0 (was a DeprecationWarning before).

- Also run notebook tests with `PIP_CONFIG_FILE=/dev/null` environment.
  Notebook tests run in a temporary virtual environment to simulate
  generic colab experience.  Here we make them more isolated from
  local pip settings.

### CODE REVIEW

**Comment by pavoljuhas** [comment]
86710bb1d6c7820497035958928cdca1607357ad addresses notebook failure in the ci-weekly test https://github.com/quantumlib/Cirq/actions/runs/20402005374

4a4f0f917824e144fa3b4cefdb823b58fcfd715d shortens test duration by about 50% if local environment is set up with an uncached airlocked repository.

File: dev_tools/notebooks/isolated_notebook_test.py [hunk 1]
```diff
@@ -155,8 +155,9 @@ def _rewrite_and_run_notebook(notebook_path, cloned_env):
         cwd=notebook_env,
         capture_output=True,
         # important to get rid of PYTHONPATH specifically, which contains
```
Comment (mhucka) [inline dev_tools/notebooks/isolated_notebook_test.py]: Trivial:

```suggestion
        # Important to get rid of PYTHONPATH specifically, which contains
```

File: dev_tools/notebooks/notebook_test.py [hunk 2]
```diff
@@ -73,7 +73,10 @@ def require_packages_not_changed() -> Iterator[None]:
 
 @pytest.fixture
 def env_with_temporary_pip_target() -> Iterator[dict[str, str]]:
-    """Setup system environment that tells pip to install packages to a temporary directory."""
+    """Setup system environment that tells pip to install packages to a temporary directory.
```
Comment (mhucka) [inline dev_tools/notebooks/notebook_test.py]:
```suggestion
    """Set up system environment that tells pip to install packages to a temporary directory.
```

**Review by mhucka — APPROVED** [review]
Trivial typos that are easy to fix. Otherwise LGTM.
